src/Administration/Resources/app/administration/src/core/data/entity-definition.data.ts

```diff
@@ -51,9 +51,20 @@ export default class EntityDefinition<EntityName extends keyof EntitySchema.Enti
 
     readonly properties: Properties;
 
-    constructor({ entity, properties }: { entity: Entity<EntityName>; properties: Properties }) {
-        this.entity = entity;
-        this.properties = properties;
+    readonly readProtected: boolean;
+
+    readonly writeProtected: boolean;
+
+    constructor(schema: {
```

Comment:
actually you don't have to use the brackets as it seems

```suggestion
    constructor({
        entity,
        properties,
        'read-protected': readProtected,
        'write-protected': writeProtected,
    }: {
        entity: Entity<EntityName>;
        properties: Properties;
        'read-protected': boolean;
        'write-protected': boolean;
    }) {
        this.entity = entity;
        this.properties = properties;
        this.readProtected = readProtected;
        this.writeProtected = writeProtected;
    }
```